### PR TITLE
PSD-1120 Improved navigation header

### PR DESCRIPTION
**Pull request by jasiekmiko**
## Description
I've ended up implementing both the govuk header and a forked version specific to psd. The former remains useful in the components gallery - perhaps it'll be helpful on cosmetics, too - @tom-bell-softwire?

I've also taken the chance to pull out some of the common html head code - again, might become useful on cosmetics.

## Checklist:

- [x] Automated checks are passing locally.
- [x] CHANGELOG updated if change is worth telling users about.
### General testing
- [x] Test without javascript
- [x] Test on small screen
### Accessibility testing
- [x] Works keyboard only
- [x] Tested with one screen reader
- [x] Zoom page to 400% - content still visible
- [x] Disable css - does content make sense and appear in a logical order?
  Note: The secondary nav is duplicated, but it's an OKed concession
- [x] Passes automated checker (automated in build or manual)

### CODE REVIEW

File: psd-web/app/views/components/_psd_header.html.slim [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  PSD has some specific navigation needs which were not implementable using the standard govuk header.
+  The macro interface has been extended, but still respects all the options as per original.
+
+  I still makes use of the govuk header module's javascript!
+
+  The big change is the addition of `secondary_navigation` items, which are displayed in a separate list when there's
+  enough real estate, but are always semantically presented as part of the nav
+
+- attributes = (local_assigns[:attributes] || {})
```
Comment (kicferk1) [inline psd-web/app/views/components/_psd_header.html.slim]: I like that way of dealing with attributes

File: psd-web/app/views/components/_psd_header.html.slim [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+              The <image> element is a valid SVG element. In SVG, you would specify
+              the URL of the image file with the xlink:href – as we don't reference an
+              image it has no effect. It's important to include the empty xlink:href
+              attribute as this prevents versions of IE which support SVG from
+              downloading the fallback image when they don't need to.
+              In other browsers <image> is synonymous for the <img> tag and will be
+              interpreted as such, displaying the fallback image.
+            - src = "#{local_assigns[:assetsPath] || '/assets/images'}/govuk-logotype-crown.png"
+            image.psd-header__logotype-crown-fallback-image src=src
+          '
```
[the remark this answers is not in the document]
Comment (jasiekmiko) [reply psd-web/app/views/components/_psd_header.html.slim]: after an infuriating search - yes, though they do it in a much less obvious way. grrrrrr!

**Comment by kicferk1** [comment]
Also, looks like you also fell into a trap of not changing cosmetics, cause we both know they don't really use our components ;) 
Aaand, some of our system tests seem to be broken :(

**Comment by jasiekmiko** [comment]
> Also, looks like you also fell into a trap of not changing cosmetics, cause we both know they don't really use our components ;)

haha not quite, but yes, pretty much. looks like BOTH projects had a tiny bit of useless code leftover from alpha ;)

**Pull request merged**